### PR TITLE
✨ Detailed Integration Testcases: Exclusion codes

### PR DESCRIPTION
**Pull request by ppratikcr7**
This PR resolves issue #1325 

Refer [document](https://docs.google.com/document/d/1e-FPHZ5ofo1u_3MSuhf7VVJdMQmQ2U9vQGf_luyoCHo/edit?usp=sharing) for test case scenarios.

### CODE REVIEW

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  checkMarkExperimentPointForUser(
+    markedExperimentPoint,
+    experimentUsers[0].id,
+    experimentName,
+    experimentPoint,
+    1
+  );
+
+  // the user should be excluded due to 'PARTICIPANT_ON_EXCLUSION_LIST' exclusion code:
+  let individualExclusions = await checkService.getAllIndividualExclusion();
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts:76]: let -> const

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  expect(individualExclusions.length).toEqual(1);
+  expect(individualExclusions).toEqual(
+    expect.arrayContaining([
+      expect.objectContaining({
+        exclusionCode: EXCLUSION_CODE.PARTICIPANT_ON_EXCLUSION_LIST
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts:81]: You should also add the user information to make sure Exclusion code is added for that user

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      })
+    ])
+  );
+
+  // delete the experiment
+  await experimentService.delete(experimentId, user, new UpgradeLogger());
+  
+  // group experiment object
+  experimentObject = ExperimentLevelExclusionExperiment2;
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts:90]: Create a separate test case for group experiment

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  checkMarkExperimentPointForUser(
+    markedExperimentPoint,
+    experimentUsers[2].id,
+    experimentName,
+    experimentPoint,
+    1
+  );
+
+  // the user should be excluded due to 'GROUP_ON_EXCLUSION_LIST' exclusion code:
+  let groupExclusions = await checkService.getAllGroupExclusions();
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts:143]: let -> const

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/GroupExperimentExclusionCode.ts [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+  const experimentService = Container.get<ExperimentService>(ExperimentService);
+  const userService = Container.get<UserService>(UserService);
+  const checkService = Container.get<CheckService>(CheckService);
+
+  // creating new user
+  const user = await userService.upsertUser(systemUser as any, new UpgradeLogger());
+
+  // First excluding a group to check "EXCLUDED_DUE_TO_GROUP_LOGIC"
+  // group experiment object
+  let experimentObject = groupLevelExclusionExperiment;
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/GroupExperimentExclusionCode.ts:23]: let -> const

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/GroupExperimentExclusionCode.ts [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+        name: experimentObject.name,
+        state: experimentObject.state,
+        postExperimentRule: experimentObject.postExperimentRule,
+        assignmentUnit: experimentObject.assignmentUnit,
+        consistencyRule: experimentObject.consistencyRule,
+      }),
+    ])
+  );
+
+  let experimentName = experimentObject.partitions[0].target;
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/GroupExperimentExclusionCode.ts:40]: let -> const

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/GroupExperimentExclusionCode.ts [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+  // the user should be excluded due to 'EXCLUDED_DUE_TO_GROUP_LOGIC' exclusion code:
+  groupExclusions = await checkService.getAllGroupExclusions();
+  expect(groupExclusions.length).toEqual(1);
+
+  individualExclusions = await checkService.getAllIndividualExclusion();
+  expect(individualExclusions.length).toEqual(2);
+  expect(individualExclusions).toEqual(
+    expect.arrayContaining([
+      expect.objectContaining({
+        exclusionCode: EXCLUSION_CODE.EXCLUDED_DUE_TO_GROUP_LOGIC
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/exclusionCode/GroupExperimentExclusionCode.ts:129]: Add user information also.

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]
Can you split the test case into multiple files with one code per file.
There are common comment across the PR

File: backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts (hunk 3 above)
Comment (ppratikcr7) [reply backend/packages/Upgrade/test/integration/Experiment/exclusionCode/ExperimentLevelExclusionCode.ts]: Done

**Comment by ppratikcr7** [comment]
> Can you split the test case into multiple files with one code per file. There are common comment across the PR

@VivekFitkariwala All peer review comments have been addressed in the last commit. Can you please re-review the PR?

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**